### DISCUSSION TITLE
Road to v1?

### DISCUSSION THREAD

**Discussion by tutts** [Adyen/adyen-react-native#136] [General]
Hello,

First, thank you for the initiative of creating an official React Native module 👏 We're really excited about it, but at a crossroad at the moment. We rolled our own RN Adyen Module about a year ago using v3 of the SDK, we want to upgrade to the latest version as we expand our International presence in the next few months, and I've been keeping a close eye on this repository for a few months, as we'd love to leverage an official module vs maintaining our own in-house.

I was wondering whether there's anyone who has this running in production right now? If so any feedback on stability? 

To the maintainers - do you have a public TODO list you're working through to get to v1?

Thanks!
Sam

**Comment by camil-adyen** [comment]
Hi @tutts,

Thank you for reaching out and nice to hear that you are considering our React Native SDK. 

We have several things on our TODO list before we officially release v1. To start, we would like to extend the SDK with an option to integrate API-only. We will do this by offering the [custom card integration](https://docs.adyen.com/payment-methods/cards/custom-card-integration) for the React Native SDK. This integration will ensure PCI compliance while using your own UI. After releasing this functionality, we will monitor the stability of the SDK and work on improvements where necessary. 

Once we're happy with the stability, we are also planning to adopt Typescript to improve developer experience. Finally, we need to add [telemetry](https://docs.adyen.com/online-payments/analytics-and-data-tracking) to the SDK to track and improve its performance. 

After completion of the above, we will release v1 with proper documentation. 

Cheers,
Camil

**Reply by tutts** [reply]
Hi @camil-adyen,

That's fantastic thank you, and interesting about the API-only version. We'll certainly take that into consideration.

Millionaire dollar question... 😆do you have any rough timelines you're looking to target? Just so we know whether to invest time in integrating/trialing the BETA vs migrating our in-house solution.

thanks!

**Reply by camil-adyen** [reply]
HI @tutts,

Good to hear! 

We're aiming to release a beta10 version with the API-only functionally in the beginning of March. Regarding the official v1 - we hope to release that mid April.

**Comment by descorp** [comment]
FYI it is moved to a **Release Candidate** mode :)

**Comment by tutts** [comment]
Hi @descorp thanks for the heads up. We're going to start integration this week and play around with it. Also just as an FYI, are you using the v5 of the Adyen SDK under the hood? Thanks!

**Reply by descorp** [reply]
Hey @tutts 

>are you using the v5 of the Adyen SDK under the hood

Unfortunately, not yet. We expecting full transition to v5 and sessions around Q3.

**Reply by tutts** [reply]
Hi @descorp, no worries, not a blocker from our side. So to confirm it's using v4?

**Reply by descorp** [reply]
Correct.

We are using badges to keep track of current version of iOS and Android SDK used.
<img width="853" alt="Screenshot 2023-03-28 at 15 20 50" src="https://user-images.githubusercontent.com/2648655/228250370-4ffb3515-e58b-4194-a76f-bb821aff5442.png">
